CONTRIBUTING_A_NEW_QUEST.md

```diff
@@ -0,0 +1,390 @@
+
+### Extras
+
+Info listed above must be supplied by every quest. But there are also several optional fields. This specific qest has
+```
+    override fun getHighlightedElements(element: Element, getMapData: () -> MapDataWithGeometry) =
+        getMapData().filter("nodes with emergency = defibrillator")
+```
+
+which causes nearby `emergency = defibrillator` nodes to be shown (icons used are defined in [PinIcons.kt](app/src/main/java/de/westnordost/streetcomplete/map/PinIcons.kt))
+
+See [also other optional properties](de/westnordost/streetcomplete/data/osm/osmquests/OsmElementQuestType.kt).
+
+### Adding quest icon
+
+It is OK to submit a quest without own icon, using any icon as a placeholder.
+
+But it would be even better to include also icon.
+
+Note that there are some not-yet-used graphics, created for some proposed or expected quests. Maybe there is no need to create icon at all!
+
+A new icon can reuse the content of [other quest icons](res/graphics/quest icons), it can be based on openly licensed graphics. See [the attribution file](res/graphics/authors.txt) for what was used so far.
+
+Keep similar style to existing ones and app in general. Once the quest icon is ready:
+
+- save as "Plain svg" or clean SVG file from unnecessary cruft in another way, like using [svgo](https://github.com/svg/svgo)
+- Put SVG into `[res/graphics/quest icons](res/graphics/quest icons)` folder
+  - SVG is a standard format editable in various software, unlike internal Android Studio xml that will be produced in the next step.
+- Open Android Studio
+- Right click on "app" folder in the Project tool window (top left)
+- Select new → vector asset
+- Select your svg file
+- Name with `ic_quest_` prefix (something like `ic_quest_traffic_calming`), this will cause IDE to generate an XML file
+- add an entry in the [attribution file](res/graphics/authors.txt)
+- modify `icon` property in the quest definition to use the new drawable
+- Commit modified or created files
+- Compile, test quest in the emulator
+
+The same method applies also to other vector drawables, only they will be placed in other parts of [res/graphics/](res/graphics/)
+
+Inkscape is a typical tool to create and edit SVG files, it is a good, free, open-source so is available to all.
+
+# Test
+
+Obviously, testing can be done also earlier. But at least before submitting for review quest should be tested.
+
+Typically it is done using an emulator. Note that you can set location in emulator settings rather than scrolling within StreetComplete itself.
+
+* Is quest listed?
+  * Look at the quest list in settings - is your quest appearing there? If not - see [this step](#Add quest to the list of active ones).
+  * While you are there you can disable all quests except yours for an easier testing.
+* Is it shown for expected elements?
+* Is it selected for some unwanted elements?
```

Comment:
```suggestion
* Is it **not** selected for some unwanted elements?
```

Reply:
I meant checking is it appearing for elements where it should not be triggered (for example asking about name for objects with name tagged because `!name` filter was missing)